taf/api/roles.py

```diff
@@ -420,7 +572,11 @@ def _get_roles_key_size(role, keystore, keys_num):
 
 
 def _role_obj(role, repository, parent=None):
-    repository = repository._tuf_repository
+    """
+    Return role TUF object based on its name
+    """
+    if isinstance(repository, Repository):
+        repository = repository._repository
```

Comment:
Will `repository` be None if repository is not an instance of `Repository`?

Reply:
TBH, this was a quick fix because repository was either a TAF repository or a TUF repository (which is what `repository._repository` returns. We should look into all calls and standardize the input (either expect one or the other). Not sure if that's worth cramming into this PR because it's mostly focused on logging and docstirng. We could create an issue to make sure we don't forget about it

Reply:
OK, let's create an issue